src/EnergyPlus/ZoneAirLoopEquipmentManager.cc

```diff
@@ -215,341 +254,191 @@ namespace ZoneAirLoopEquipmentManager {
-                        DataDefineEquip::ZnAirLoopEquipType::SingleDuct_ConstVol_4PipeInduc;
-                    if (state.dataDefineEquipment->AirDistUnit(AirDistUnitNum).UpStreamLeak ||
-                        state.dataDefineEquipment->AirDistUnit(AirDistUnitNum).DownStreamLeak) {
-                        ShowSevereError(
-                            state,
-                            format("Error found in {} = {}", CurrentModuleObject, state.dataDefineEquipment->AirDistUnit(AirDistUnitNum).Name));
-                        ShowContinueError(state,
-                                          format("Simple duct leakage model not available for {} = {}",
-                                                 cAlphaFields(3),
-                                                 state.dataDefineEquipment->AirDistUnit(AirDistUnitNum).EquipType(AirDistCompUnitNum)));
-                        ErrorsFound = true;
-                    }
-                } else if (UtilityRoutines::SameString(state.dataDefineEquipment->AirDistUnit(AirDistUnitNum).EquipType(AirDistCompUnitNum),
-                                                       "AirTerminal:SingleDuct:VAV:Reheat:VariableSpeedFan")) {
-                    state.dataDefineEquipment->AirDistUnit(AirDistUnitNum).EquipTypeEnum(AirDistCompUnitNum) =
-                        DataDefineEquip::ZnAirLoopEquipType::SingleDuctVAVReheatVSFan;
-                    if (state.dataDefineEquipment->AirDistUnit(AirDistUnitNum).UpStreamLeak ||
-                        state.dataDefineEquipment->AirDistUnit(AirDistUnitNum).DownStreamLeak) {
-                        ShowSevereError(
-                            state,
-                            format("Error found in {} = {}", CurrentModuleObject, state.dataDefineEquipment->AirDistUnit(AirDistUnitNum).Name));
+                switch (airDistUnit.EquipTypeEnum(AirDistCompUnitNum)) {
+                case DataDefineEquip::ZnAirLoopEquipType::DualDuctConstVolume:
+                case DataDefineEquip::ZnAirLoopEquipType::DualDuctVAV:
+                case DataDefineEquip::ZnAirLoopEquipType::DualDuctVAVOutdoorAir:
+                case DataDefineEquip::ZnAirLoopEquipType::SingleDuct_SeriesPIU_Reheat:
+                case DataDefineEquip::ZnAirLoopEquipType::SingleDuct_ParallelPIU_Reheat:
+                case DataDefineEquip::ZnAirLoopEquipType::SingleDuct_ConstVol_4PipeInduc:
+                case DataDefineEquip::ZnAirLoopEquipType::SingleDuctVAVReheatVSFan:
+                case DataDefineEquip::ZnAirLoopEquipType::SingleDuctConstVolCooledBeam:
+                case DataDefineEquip::ZnAirLoopEquipType::SingleDuctUserDefined:
+                case DataDefineEquip::ZnAirLoopEquipType::SingleDuctATMixer:
+                    if (airDistUnit.UpStreamLeak || airDistUnit.DownStreamLeak) {
+                        ShowSevereError(state, format("Error found in {} = {}", CurrentModuleObject, airDistUnit.Name));
                         ShowContinueError(state,
```

Comment:
Replaced `IF logic` with `switch` using the enum class, air terminal string_view array, and getEnumValue() function.